provisioning/transport/amqp/src/AmqpLinkFactory.cs

```diff
@@ -0,0 +1,56 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Amqp;
+
+namespace Microsoft.Azure.Devices.Provisioning.Client.Transport.Amqp
+{
+    class AmqpLinkFactory : ILinkFactory
+    {
+        public IAsyncResult BeginOpenLink(AmqpLink link, TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            return TaskHelpers.ToAsyncResult(this.OpenLinkAsync(link, timeout), callback, state);
+        }
+
+        public AmqpLink CreateLink(AmqpSession session, AmqpLinkSettings settings)
+        {
+            AmqpLink link;
+            if (settings.IsReceiver())
+            {
+                link = new ReceivingAmqpLink(session, settings);
+            }
+            else
+            {
+                link = new SendingAmqpLink(session, settings);
+            }
+            this.OnLinkCreated(link);
+            return link;
+        }
+
+        public void EndOpenLink(IAsyncResult result)
```

Comment:
Not used.

Reply:
interface member

---
In reply to: [146718122](https://github.com/Azure/azure-iot-sdk-csharp/pull/204#discussion_r146718122) [](ancestors = 146718122)